clang/lib/Tooling/DependencyScanning/DependencyScanningWorker.cpp

```diff
@@ -586,6 +586,11 @@ class DependencyScanningAction : public tooling::ToolAction {
     auto *FileMgr = ScanInstance.createFileManager(FS);
     ScanInstance.createSourceManager(*FileMgr);
 
+    // Initialize PrefixMapper incase mappings exist.
+    DepscanPrefixMapping::configurePrefixMapper(
```

Comment:
I don't necessarily see a problem with using ScanInstance.PrefixMapper, but configuring it again here seems like it creates tension with the scanning action controller, which otherwise controls this behaviour.  We could modify `CASFSActionController::initialize` and `IncludeTreeActionController::initialize` to call `setPrefixMapper` on the scanning instance instead.

Reply:
I like that better, too. Right now, the PCH dependency collection & `ModuleDepCollector` creation happen before the controller is initialized. I am looking into whether it's reasonably simple to change the order.